chasm/search_attribute.go

```diff
@@ -382,3 +384,167 @@ func (s SearchAttributeKeywordList) Value(value []string) SearchAttributeKeyValu
+	}
+
+	return string(stringValue), true
+}
+
+// GetDateTime returns the time value for a given SearchAttributeDateTime. If not found or map is nil, second parameter is false.
+func (m SearchAttributesMap) GetDateTime(sa SearchAttributeDateTime) (time.Time, bool) {
+	if m.values == nil {
+		return time.Time{}, false
+	}
+
+	alias := sa.definition().alias
+	timeValue, ok := m.values[alias].(VisibilityValueTime)
+	if !ok {
+		return time.Time{}, false
+	}
+
+	return time.Time(timeValue), true
+}
+
+// GetKeywordList returns the string list value for a given SearchAttributeKeywordList. If not found or map is nil, second parameter is false.
+func (m SearchAttributesMap) GetKeywordList(sa SearchAttributeKeywordList) ([]string, bool) {
+	if m.values == nil {
+		return nil, false
+	}
+
+	alias := sa.definition().alias
+	keywordListValue, ok := m.values[alias].(VisibilityValueStringSlice)
+	if !ok {
+		return nil, false
+	}
+
+	return []string(keywordListValue), true
+}
+
+// convertToVisibilityValue converts a value to VisibilityValue based on its runtime type.
+func convertToVisibilityValue(value interface{}) VisibilityValue {
+	switch val := value.(type) {
+	case int:
+		return VisibilityValueInt64(int64(val))
+	case int32:
+		return VisibilityValueInt64(int64(val))
+	case int64:
+		return VisibilityValueInt64(val)
+	case float32:
+		return VisibilityValueFloat64(float64(val))
+	case float64:
+		return VisibilityValueFloat64(val)
+	case bool:
+		return VisibilityValueBool(val)
+	case time.Time:
+		return VisibilityValueTime(val)
+	case string:
+		// Try to parse as datetime first
+		if parsedTime, err := time.Parse(time.RFC3339, val); err == nil {
+			return VisibilityValueTime(parsedTime)
+		}
+		return VisibilityValueString(val)
+	case []byte:
+		return VisibilityValueByteSlice(val)
+	case []string:
+		return VisibilityValueStringSlice(val)
+	default:
+		// Return as string if type is unknown
+		return VisibilityValueString(fmt.Sprintf("%v", val))
+	}
+}
+
+// AliasChasmSearchAttributes converts search attribute values to VisibilityValue and aliases field names.
+// It takes a map of field names to interface{} values, converts them to VisibilityValue based on their runtime type,
+// and then aliases the field names using the mapper.
+func AliasChasmSearchAttributes(
+	chasmSearchAttributes map[string]interface{},
+	mapper *VisibilitySearchAttributesMapper,
+) (map[string]VisibilityValue, error) {
+	if len(chasmSearchAttributes) == 0 {
+		return nil, nil
+	}
+
+	chasmSAs := make(map[string]VisibilityValue, len(chasmSearchAttributes))
+	for fieldName, value := range chasmSearchAttributes {
+		visibilityValue := convertToVisibilityValue(value)
+		aliasName, err := mapper.Alias(fieldName)
+		if err != nil {
+			// Silently ignore serviceerror.InvalidArgument because it indicates unmapped field, search attribute is not registered.
+			// IMPORTANT: AliasChasmSearchAttributes InvalidArgument indicates a bug in the code, not a user error.
```

Comment:
Not necessarily a bug in the code, it could be that the search attribute was deleted.